src/python/pants/backend/python/subsystems/python_repos.py

```diff
@@ -4,10 +4,57 @@
 
 from __future__ import absolute_import, division, print_function, unicode_literals
 
+import logging
+
 from pex.fetcher import Fetcher, PyPIFetcher
-from pex.http import Context
+from pex.http import RequestsContext, StreamFilelike, requests
 
 from pants.subsystem.subsystem import Subsystem
+from pants.util.memo import memoized_method
+
+
+logger = logging.getLogger(__name__)
+
+
+_REQUESTS_TIMEOUTS = (15, 30)
+
+
+def _open_monkey(self, link):
+  # requests does not support file:// -- so we must short-circuit manually
+  if link.local:
+    return open(link.local_path, 'rb')  # noqa: T802
+  for attempt in range(self._max_retries + 1):
```

Comment:
cant see it here, but assuming `_max_retries` is being set somewhere else

Reply:
Yea, `self` in this case will be a RequestsContext instance.